product_variant_default_code/models/product.py

```diff
@@ -76,19 +76,51 @@ class ProductTemplate(models.Model):
         "attribute name",
     )
 
-    @api.depends(
-        "code_prefix",
-        "attribute_line_ids",
-        "attribute_line_ids.attribute_id.name",
+    variant_default_code_error = fields.Text(
+        compute="_compute_variant_default_code_error"
     )
-    def _compute_reference_mask(self):
+
+    def is_automask(self):
         automask = bool(
             not self.user_has_groups(
                 "product_variant_default_code.group_product_default_code_manual_mask"
             )
         )
+        return automask
+
+    @api.depends(
+        "code_prefix",
+        "attribute_line_ids.value_ids",
+        "attribute_line_ids.value_ids.code",
+        "attribute_line_ids.value_ids.name",
+    )
+    def _compute_variant_default_code_error(self):
+        if self.is_automask():
+            for rec in self:
+                error_txt = ""
+                if not rec.code_prefix and self.is_automask():
+                    error_txt += "Reference Prefix is missing.\n"
+                values = self.attribute_line_ids.value_ids
+                if not all(values.mapped("code")):
+                    error_txt += "Some attribute codes are missing:\n"
+                    for value in values:
+                        if not value.code:
+                            error_txt += "- The code of the attribute value"
+                            error_txt += " '{}' is missing\n".format(value.name)
+                if error_txt:
+                    error_txt = "".join(
+                        ("Default Code can not be computed.\n", error_txt)
+                    )
+                rec.variant_default_code_error = error_txt
+
+    @api.depends(
+        "code_prefix",
+        "attribute_line_ids",
+        "attribute_line_ids.attribute_id.name",
+    )
+    def _compute_reference_mask(self):
         for rec in self:
-            if automask or rec.reference_mask == "":
+            if self.is_automask() or rec.reference_mask == "":
```

Comment:
call is_automask before the loop, it will avoid to restest the group for each record